### DISCUSSION TITLE
Build process

### DISCUSSION THREAD

**Discussion by aswindevaraj19** [langroodi/Adaptive-AUTOSAR#18] [Q&A] [answered]
The build process is not clear.
Can you elaborate on it???

**Comment by langroodi** [comment, marked as answer]
The build will be done via the CMake file. So, be sure first that you have already installed CMake (version 3.14 or higher) and one of GCC 11 or Clang 14 compiler on your Linux.

Then you need to configure the compiler. So, open the terminal and use `cd` to go to the repository clone directory and if you prefer to use GCC, run the following command on your terminal:
```
cmake -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/x86_64-linux-gnu-gcc-11 -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/x86_64-linux-gnu-g++-11 -S . -B build
```

and if you prefer Clang, run the following command on your terminal:
```
cmake -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/clang-14 -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/clang++-14 -S . -B build
```

in which:

- `-DCMAKE_BUILD_TYPE` sets the build type to `Debug` which enables you to debug the application after the build;
- `-DCMAKE_C_COMPILER` sets the C language compiler path which is required for kernel function calls and the libcurl;
- `-DCMAKE_CXX_COMPILER:FILEPATH` sets the C++ language compiler path for rest of the application;
- `-S` switch sets the current directory as the source directory;
- `-B` switch creates a `build` directory within the current directory for exporting the build results.

Then you need to compile the source via running `cmake --build build` in your terminal. And after couple of minutes, you can expect your build results in the `build` folder.

**Reply by RajeshPolamReddy** [reply]
Once the build is completed 
there is no ./build/adaptive_autosar: No such file or directory

How to we run the project 
**./build/adaptive_autosar .**/configuration/execution_manifest.arxml ./configuration/extended_vehicle_manifest.arxml ./configuration/diagnostic_manager_manifest.arxml ./configuration/health_monitoring_manifest.arxml

How to get /build/adaptive_autosar: ? can you please help

**Reply by RajeshPolamReddy** [reply]
I have this in build/bin/adaptive_autosar 
May be the run command to be updated ?

**Reply by langroodi** [reply]
It can happen due to your build chain configuration or the command you run to build the project, the executable ends up in `build/bin` instead of `build` folder. And in that case as you said, the run command should be:
```shell
./build/bin/adaptive_autosar ./configuration/execution_manifest.arxml ./configuration/extended_vehicle_manifest.arxml ./configuration/diagnostic_manager_manifest.arxml ./configuration/health_monitoring_manifest.arxml
```
However, because this is the specific case, I cannot update the run command in the readme file. As an example, the build pipeline follows the same command mentioned in readme, and the executable ends up in the `build` folder instead of `build/bin`.

**Reply by RajeshPolamReddy** [reply]
Okay got it 
Thanks for quick clarification
